openscm/scmdataframe.py

```diff
@@ -6,9 +6,13 @@
 
 
 class ScmDataFrameBase(IamDataFrame):
-    """This base is the class other libraries can subclass
+    """
+    This base is the class other libraries can subclass.
 
-    Having such a subclass avoids a potential circularity where e.g. openscm imports ScmDataFrame as well as Pymagicc, but Pymagicc wants to import ScmDataFrame and hence to try and import ScmDataFrame you have to import ScmDataFrame itself (hence the circularity).
+    Having such a subclass avoids a potential circularity where e.g. openscm imports
+    ScmDataFrame as well as Pymagicc, but Pymagicc wants to import ScmDataFrame and
+    hence to try and import ScmDataFrame you have to import ScmDataFrame itself (hence
+    the circularity).
```

Comment:
@swillner can you put this back in a separate file please? The comment above should explain why.

Reply:
sounds very much like a problem in pymagicc that create the circular dependencies. but will split the file again.

Reply:
> will split the file again.

Thank you

At the moment I think it's a problem for any adapter package that also wants to provide an OpenSCM interface directly. The reason is:

- OpenSCM is imported
- that imports `openscm.highlevel`
- which imports `openscm.adapters`
- which imports `<model-adapter-package>`, e.g. `pymagicc`
- which imports `openscm.highlevel` in order to get `ScmDataFrame`

and hence your circularity. Having said that, it might be solved thanks to the lazy imports of adapters because that means that you've already loaded `openscm.highlevel` when you load `<model-adapter-package>`. I'll try putting it back in one file when I add the MAGICC adapter and see what happens.